### PR TITLE
Increase psalm level to 2

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | -

### CODE REVIEW

File: src/Paginator/KeysetPaginator.php [hunk 1]
```diff
@@ -257,7 +281,7 @@ private function getReverseFilter(Sort $sort): CompareFilter
         return new GreaterThanOrEqual($field, $this->getValue());
     }
 
-    private function getValue(): string
+    private function getValue(): ?string
```
Comment (roxblnfk) [inline src/Paginator/KeysetPaginator.php:284]: Can you add test for this case?

File: src/Paginator/KeysetPaginator.php [hunk 2]
```diff
@@ -32,13 +40,21 @@
 class KeysetPaginator implements PaginatorInterface
 {
     /**
-     * @var FilterableDataInterface|ReadableDataInterface|SortableDataInterface
```
[the remark this answers is not in the document]
Comment (vjik) [reply src/Paginator/KeysetPaginator.php:35]: Returned.

File: src/Paginator/KeysetPaginator.php [hunk 3]
```diff
@@ -32,13 +40,21 @@
...
+     * @psalm-var DataReaderType
      */
     private ReadableDataInterface $dataReader;
     private int $pageSize = self::DEFAULT_PAGE_SIZE;
     private ?string $firstValue = null;
     private ?string $lastValue = null;
+
+    /**
+     * @var mixed
```
[the remark this answers is not in the document]
Comment (vjik) [reply src/Paginator/KeysetPaginator.php]: I did a refactoring and now use `?string`.

File: src/Paginator/KeysetPaginator.php [hunk 4]
```diff
@@ -32,13 +40,21 @@
...
+     */
     private $currentFirstValue;
+
+    /**
+     * @var mixed
```
[the remark this answers is not in the document]
Comment (vjik) [reply src/Paginator/KeysetPaginator.php]: I did a refactoring and now use `?string`.

File: src/Paginator/KeysetPaginator.php (hunk 1 above)
Comment (vjik) [reply src/Paginator/KeysetPaginator.php]: I'm returned `string` and add comment for `@psalm-suppress`.

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
👍

**Pull request merged**